core/src/main/java/io/grpc/Metadata.java

```diff
@@ -490,6 +491,7 @@ private boolean bytesEqual(byte[] left, byte[] right) {
   }
 
   /** Marshaller for metadata values that are serialized into raw binary. */
+  @Immutable
```

Comment:
The metadata is supposed to be transmitted between server and client, it might be unsafe or errorprone if the marshaller is able to change settings over time. It make sense in some scenario, the marshaller might need be stateful, but it may be rare and it's better to be defensive.